### PR TITLE
[Mobile] Cover: Add Media Button when only colour is selected

### LINKED ISSUE #23877: [Mobile] Cover: Add Media Button when only colour is selected
**Is your feature request related to a problem? Please describe.**
The user should have an easy way to add a media background when only color is set

**Describe the solution you'd like**
Add a image icon button should be added as depicted bellow

83098993-a187c680-a071-11ea-83d4-4ae56f6c5850

**Describe alternatives you've considered**
There is already an alternative way to add media by long pressing on the cover block

**Additional Context**
This issue is part of breaking down issue **https://github.com/wordpress-mobile/gutenberg-mobile/issues/2275** into smaller subtasks. Opening the new tickets here as part of the GB-Mobile Monorepo migration.

### PR DESCRIPTION
**Pull request by antonis**
**Fixes** https://github.com/WordPress/gutenberg/issues/23877 (**parent** https://github.com/wordpress-mobile/gutenberg-mobile/issues/2275)

## Related PRs:
This PR depends on https://github.com/WordPress/gutenberg/pull/23870

* `gutenberg-mobile` https://github.com/wordpress-mobile/gutenberg-mobile/pull/2513
* **Depends on** Create cover block with custom colour https://github.com/WordPress/gutenberg/pull/23870
* Create cover block with only colour from a default palette https://github.com/WordPress/gutenberg/pull/23833
* Clear media button in block settings https://github.com/WordPress/gutenberg/pull/23794

## Description
An image icon button has been added at the top-left of the component. When pressed the add media options are presented.

## How has this been tested?
* Add a cover block
* Press on a colour from the palette
* Verify that a new cover block with only colour is created
* Verify that the new icon is presented at the top-left of the component
* Tap on the icon and verify that the add media options are presented
* Add a media and verify that it is applied in the background

## Screenshots

|Screenshot|Functionality|
|---|---|
|Simulator Screen Shot - iPhone 11 - 2020-07-10 at 18 51 16|app|

## Types of changes
Enhancement

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by antonis** [comment]
Hello @iamthomasbishop 👋 ,
Please have a look at the media button addition and let me know wdyt.

**Comment by iamthomasbishop** [comment]
Hi @antonis, can we use the same button style/sizing/spacing as is used on the edit button when a media item is attached (with a different icon, of course)? Example:

image

**Comment by antonis** [comment]
Thank you for the feedback @iamthomasbishop :)

I updated the button design and changed the text label.
Since the MediaUpload bottom sheet component already had a text for this I kept it "Choose image or video"

"Light background|Dark background|
|---|---|
|Simulator Screen Shot - iPhone 11 - 2020-07-23 at 10 10 29|Simulator Screen Shot - iPhone 11 - 2020-07-23 at 10 05 25|

**Comment by iamthomasbishop** [comment]
> I updated the button design and changed the text label.
> Since the MediaUpload bottom sheet component already had a text for this I kept it "Choose image or video"

Perfect. Thank you @antonis !

**Review by mkevins — COMMENTED** [review]
This is looking pretty good. I tested on Android, and the logic is working as expected. There is one issue with the button size / container size that makes tapping difficult. It seems that the tappable area is small, and offset from the button itself:



I inspected the layout and it seems that the views are rendered outside their container (so they don't capture the event). E.g. the red rectangle is the outer view and the blue one is the child view:
add-image-button-out-of-container

It seems that aside from being offset, the size of the touchable is that of the inner view (inner view, e.g. icon, is the red rectangle):
add-image-button-inner-view

I tried locally changing the order (so `TouchableWithoutFeedback` wraps `View`), but the problem still persisted, so perhaps a change is needed elsewhere, either instead of or in addition to that.

File: packages/block-library/src/cover/edit.native.js [hunk 1]
```diff
@@ -330,6 +346,15 @@ const Cover = ( {
 		</InspectorControls>
 	);
 
+	const renderContent = ( getMediaOptions ) => (
+		<>
+			{ renderBackground( getMediaOptions ) }
+			{ isParentSelected &&
+				hasOnlyColorBackground &&
+				addMediaButton( openMediaOptionsRef.current ) }
```
[the remark this answers is not in the document]
Comment (antonis) [reply packages/block-library/src/cover/edit.native.js]: Hello @mkevins 👋 
thank you for your feedback, I really appreciate it :)
You are right, the parameter can be accessed directly. I updated the PR changing this and fixing the tappable area.

File: packages/block-library/src/cover/edit.native.js [hunk 2]
```diff
@@ -252,9 +252,12 @@ const Cover = ( {
 		</BlockControls>
 	);
 
-	const addMediaButton = ( open ) => (
+	const addMediaButton = () => (
 		<View style={ styles.selectImageContainer }>
```
Comment (mkevins) [inline packages/block-library/src/cover/edit.native.js:256]: We could remove the double `styles.selectImageContainer` by wrapping the view like this:

```patch
diff --git a/packages/block-library/src/cover/edit.native.js b/packages/block-library/src/cover/edit.native.js
index 14b413712a..d698791df7 100644
--- a/packages/block-library/src/cover/edit.native.js
+++ b/packages/block-library/src/cover/edit.native.js
@@ -253,11 +253,8 @@ const Cover = ( {
 	);
 
 	const addMediaButton = () => (
-		<View style={ styles.selectImageContainer }>
-			<TouchableWithoutFeedback
-				style={ styles.selectImageContainer }
-				onPress={ openMediaOptionsRef.current }
-			>
+		<TouchableWithoutFeedback onPress={ openMediaOptionsRef.current }>
+			<View style={ styles.selectImageContainer }>
 				<View style={ styles.selectImage }>
 					<Icon
 						size={ 16 }
@@ -265,8 +262,8 @@ const Cover = ( {
 						{ ...styles.selectImageIcon }
 					/>
 				</View>
-			</TouchableWithoutFeedback>
-		</View>
+			</View>
+		</TouchableWithoutFeedback>
 	);
 
 	const controls = (

```

**Review by mkevins — APPROVED** [review]
Looks good and it's working correctly now. Nice work @antonis ! :tada: 

I left one minor suggestion (as a patch, since the GitHub diff context couldn't capture it as a "one-click-suggestion"), not a blocker though.

File: packages/block-library/src/cover/edit.native.js (hunk 2 above)
Comment (antonis) [reply packages/block-library/src/cover/edit.native.js]: Thank you for the suggestion @mkevins! This looks much better. I updated the PR

**Pull request merged**